### DISCUSSION TITLE
Very Slow Download Speed

### DISCUSSION THREAD

**Discussion by NightAngel** [aclap-dev/video-downloadhelper#698] [Help and Q&A]
### Your question

I think the app downloading speed is getting so very slow... 300 MB download took over 20 mins. If the file is bigger than 300Mb then it takes nearly an hour.  This is not quite right.. and it happens on every site, like VK for example. Last Sunday it was fine..so fast like I just take a blink and it's done. I turn my computer on again this Sunday and the downloading speed is so terrible. I even called the internet company to check if there's anything wrong with my internet. I have to switch to another download app for now. 

I've search for this problem but it seems like there's no answer yet. Lots of people experiencing the same problem here. I really hope it can be fix real soon. It was such a great app. 

### Addon & CoApp version and details

version: 9.0.1.22
target: google
channel: stable
lang: en-US
coapp: {"found":true,"new_version":false,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19"}
license: {"accepted":true,"email":"<removed>","key":"81279b6d49114631"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36
{
    "valid_license_message_has_been_displayed": true,
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\Com\\Downloads",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "F:\\Series\\"
        }
    },
    "concurrent_downloads_max": 8,
    "never_show_no_incognito_msg_again": true,
    "theme": "dark",
    "view_options": {
        "all_tabs": false,
        "hide_downloaded": false,
        "low_quality": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "sort_by_status": true,
        "sort_reverse": false
    },
    "use_sidebar": false,
    "open_count_store": 62,
    "successfull_dl": 23
}

**Comment by GrampaWildWilly** [comment]
In general, download speed is a function of your ISP & the serving web site.  Servers can be fast today & slow tomorrow.  It happens.  But I can test something if you give me a specific URL of a page on which a video is hosted.  Don't tell me a web site.  Point me to a video.  Or a few, on different web sites, if you think that would clarify the issue.  Let's hope I don't get geoblocked on anything.

**Reply by NightAngel** [reply]
I will check for more. I understand that some sites are slow sometimes.. Since yesterday every sites I visit and downloaded from are all slow. Maybe some other sites that I've never been to are not that slow. But there are also some other users who experiencing the same problem so, I'll wait for another week to see if it's fix. Thank you for your reply.

**Comment by NightAngel** [comment]
So, I visited some other websites and try downloaded some videos. If it's not over 300MB the download speed is still okay. Over than that It's pretty slow. And I just got the brand new router from the internet company because I called and tell them my internet speed is slow. Actually.. It's only slow when I use video download helper... When I use IDM it works just fine...

**Reply by paulrouget** [reply]
Can you give me one specific website where you experience the download speed issue?

**Reply by NightAngel** [reply]
https://anime-indy.com/watch/one-hundred-thousand-years-of-qi-refining-ss-episode-116-sub/

I just downloaded this. The file is 185 MB, started time 08.05 pm. Finished Time 08.11 pm. It's not that I can't wait but it's way faster before. The file this size usually took not over 2 mins to download last week. 

Here's another one  https://fmoviesz.to/tv/csi-vegas-kx0qr/2-11
Starting at 08.14 pm. Finished at 08.28 pm. File size 313 MB
I understand that these movies sites are quite slow sometimes. But it's been like this for the past two days. When I use IDM it's fine. 

And I also got "Download Failed" today too from another movies site. I just downloaded the series file from that site this afternoon but now the app failed to download from it.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3593, Firefox 126.0 64-bit, licensed VDH 9.0.2.3a1 beta, CoApp 2.0.19.

For your first URL, it made my eyeballs spin in their sockets.  It's about the busiest web page I've ever seen.  It's in some language I don't speak.  My best guesses are that it's either Sanskrit or Thai.  But I couldn't make heads or tails of the page so I just closed it.  Moving on . . .

Your second URL nearly defeated me as well, between incessantly nagging  me to install the Opera web browser, ads that insisted they couldn't be displayed embedded in that page but needed to be launched in their own separate page, pages blocked as threats by my ISP, I did finally manage to launch playback of that episode of CSI.  It was not until I did launch it that anything showed up in the VDH menu.  It downloaded at a very good speed.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/988fdb05-e5fd-4e8b-8884-b3814160b6cc)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f1df3513-5e6b-4df2-96b0-23f22dde375f)

7.6 million bytes per second is above the average of what I see in my own downloads & in downloads I do on behalf of our fellow users.  This is what I got:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f936fd1e-a553-4801-b16b-3f251ac4ecf5)

You won't be surprised to learn that it played just fine in VLC.  I didn't sit & watch it, just skimmed it, sampling it at intervals to make sure it was good.

For extra credit . . .

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/fd850579-ac53-4ae7-a373-b7b823dc6c6b)

The vtt object above that one looked like Italian subtitles.  The one immediately below was a collection of images.  I think there is a format of captions that follows that convention but I'm not sure what it's called.  The fourth vtt object was complete gibberish to me.  Again, it may be a format of subtitles I'm simply not familiar with.  It's possible they are captions in some language like Japaness/Korean/Chinese that requires double-byte character encoding.  I admit I'm making wild-ass guesses here.

In any case, I downloaded the English subtitles shown in that last image.

[CSI Vegas 2021 vtt Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/15463177/CSI.Vegas.2021.vtt.Log.txt)

Here's what they looked like:

![#05](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/d02dc289-a904-4dac-a4ac-b073995e1f02)

If you study my ffmpeg log carefully, you'll note that the input URL is the vtt object for the English captions from the Network Monitor.  But I specified an output file name with the extension .vtt.  That tells ffmpeg to convert the srt input into proper vtt output.  Note the absence of sequence numbers on the individual subtitles.  This is the distinguishing feature of vtt that, in my opinion, makes it superior to srt.  In my experience, if you compare srt & vtt caption files of the same dialog, the vtt file is at least 1/3 smaller.  I have seen cases in which the vtt file was less than half the size of the original srt file.

Whatever.  Here I'm playing the MP4 I downloaded with VDH with the subtitles I downloaded with ffmpeg:

![#06](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6616ed0f-0c5f-4d8c-8413-e033b276c5e9)

So.  Speed.  I'm not seeing the problem you are seeing.  This is typical.  People report speed problems here every so often.  I think the evidence is that VDH is not the culprit.  Once again I say download speed is a function of your ISP & the particular server.  Internet service is not uniformly of a given quality everywhere.  That's just the way of things.  You say an older version of VDH ran faster.  Perhaps.  I never saw speed problems I could attribute to VDH on any of the older versions.  I used to have a crummy old DSL connection that wouldn't give me anything faster than about 550,000 bytes per second.  Past versions of VDH would download at that lousy speed.  A couple of years ago, I finally ditched my local phone company in favor of my cable TV service.  My new Internet connection gives me 62.5 million bytes per second, more than 100 times faster, although only a couple of sites I visit regularly actually use all of that.  Your fmoviesz site gave me over 7 million bytes per second.  Note that.  My capacity is 62.5 million but my download was 7+ million.  That's the impact of the server throttling its service.

I'm not seeing a speed problem with the very latest version of VDH.  What conclusion can you draw?

On the subject of works now but fails later, I've been encountering sites like that lately.  Pornhub & missav come to mind.  This appears to be a problem with the web site.  It will work fine, hours later it will fail, hours later no problems.  There appear to be sites like that.  It's the web sites.  When they decide to provide reliable service, VDH gets the content without problems.  When VDH starts failing, you just have to either retry incessantly until it works, which appears to be the necessary technique on Pornhub & missav, or just wait a few hours.  There are problem reports in here of exactly that approach being required.  It's infuriating, maddening, but reality.  The problem is not in VDH.

**Reply by NightAngel** [reply]
Wow.. That's a really really long one... I think it varies for each websites and/or countries, internet services and etc. 

For me I still have to say that my VDH is getting slower. I use IDM too and it's as fast as it's always been. If I have to wait for over 10 mins for a file under 300 MB size then that's slow.

I'm not complaining that VDH is the problem. I just wanna share my situation and see if anyone experienced the same thing. But then again like I said It varies. I'm not an expert or anything.. I just need to confirm if this happens to any other users.. If you said that yours works fine then I have to accept that. 

And about the failed download, I'm not worried bout that one bit. I can re-download it anytime.. If that site doesn't work I can still find some other sites.. My one and only concern is that my VDH is getting slow... But that's not a big problem. I can use other app to download. But since I've decided to purchased VDH because I tried it and it was so fast before but then turn slow. This makes me feel a bit annoyed. 

Thank you for your effort anyway.

**Comment by Mattmjs** [comment]
A lot of sites don't give you the full download speed of your connection, to a certain degree it is throttled.
But they do provide different sever options and some are faster than others which you should experiment with.
Have you noticed those options, I took a screenshot for each site.

You also get an accurate measure of speed in the task manager window by looking at the processes tab, network column
for your browser.

anime site
![Screenshot 2024-05-28 at 16-23-18 One Hundred Thousand Years Of Qi Refining ข้าก็แค่ฝึกวิชาหนึ่งแสนปี ตอนที่ 116 ซับไทย - ANIMEiNDY](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/2e50a13b-7293-4a34-8661-74be2a11a753)

csi
![Screenshot 2024-05-28 at 16-24-20 Watch Online CSI Vegas 2021 - FMovies](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/b86bfd03-e2a2-4eb2-95f0-db6a6feb8375)

**Reply by NightAngel** [reply]
Yes I tried that too. I can be very persistent sometimes. So, I tried everything and every way I can. And it's slower than before. Maybe it's just me. Or it's the sites that change something on it.
Anyway thank you for your help.
